### DISCUSSION TITLE
Mysterious behaviour (a bug?) of GitHub Pages on a project created from a template

### DISCUSSION THREAD

**Discussion by sneas** [community/community#22068] [Other Feature Feedback, Questions, & Ideas]
<p>I’ve been experimenting with that for a day, and here is what I’ve discovered.</p>
<p>I’ve created a Template Project (<a href="https://github.com/sneas/cv-template" rel="nofollow noopener">https://github.com/sneas/cv-template</a>), which contains a GitHub Action. The action generates an HTML website and pushes it to <code>gh-pages</code> branch.</p>
<p>As an experiment, I created a new public project from that template (<a href="https://github.com/sneas/cv-4" rel="nofollow noopener">https://github.com/sneas/cv-4</a>). The action had been triggered and a new <code>gh-pages</code> branch had been created and pushed. Unfortunately, the GH page has not been generated.</p>
<p>No matter what I do (mostly - retrigger the action by pushing new code to master) - the GH Page doesn’t show up.</p>
<p>The only way to make the page show up is to go to the project’s settings, change GH Pages source branch to <code>master branch</code>, and then change it back to <code>gh-pages</code> branch.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/9892f85e6cd850579fbe02299637ce24a800f79c.png" data-download-href="/uploads/short-url/lLJsnmj3Nv3nGu2PPAP5xa0eL3m.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/144651/181068767-58a3a979-ec54-4911-9ceb-2d6b84b1bdb9.png" alt="image" data-base62-sha1="lLJsnmj3Nv3nGu2PPAP5xa0eL3m" width="690" height="382" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/9892f85e6cd850579fbe02299637ce24a800f79c_2_690x382.png, https://aws1.discourse-cdn.com/github/optimized/2X/9/9892f85e6cd850579fbe02299637ce24a800f79c_2_1035x573.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/9/9892f85e6cd850579fbe02299637ce24a800f79c_2_1380x764.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/9892f85e6cd850579fbe02299637ce24a800f79c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1876×1040 107 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>This behavior is looking counterintuitive to me, that’s why I assume it’s a bug. Is it?</p>

**Comment by tmetz** [comment, marked as answer]
<p>Hi there! :wave: Welcome to the Community!</p>
<p>You have run into a quirk of using Actions to build Pages that is coming up more and more often.</p>
<p>In order to be built and published for the very first time, a GitHub Pages repository must have a commit pushed to the publishing source branch by a user with admin permissions for the repository, such as the repository owner.</p>
<p>In these cases, the <code>github-actions</code> user pushes, but doesn’t have admin access to the repository.</p>
<p>Once a commit has been pushed to the <code>gh-pages</code> branch by a user with admin permissions, you should find the site live and available shortly thereafter. Subsequent commits pushed by accounts with write access (like <code>github-actions</code>) will then also trigger a build.</p>
<p>This is on our engineers’ radar, so stay tuned for future changes which may fix this quirk.</p>
<p>I hope that helps to explain! Let us know if there’s anything else we can do.</p>

**Comment by sneas** [comment]
<p>Hi, <a class="mention" href="/tmetz">@tmetz</a>! Thank you for your reply. It’s a pity that a site couldn’t be created automatically with github-actions. Guess, I’ll just have to update my documentation to fix the problem for now.</p>
